bin/process.sh

```diff
@@ -6,10 +6,15 @@ shift $[$#-2]
 
 input=$1
 output=$2
-# target size in KB
-THUMBNAIL_SIZE=${THUMBNAIL_SIZE:-300}
+
+THUMBNAIL_SIZE=${THUMBNAIL_SIZE:-300} # target size in KB
 TILER_BASE_URL=${TILER_BASE_URL:-http://tiles.openaerialmap.org}
 
+# Although TMPDIR is usually set by the OS, it doesn't seem to be getting passed through
+# Docker and Node's `spawn()`, so no harm done in just defaulting to `/tmp` in such cases.
+# There may be clues here as the underlying issue: https://github.com/npm/npm/issues/4531
+TMPDIR={TMPDIR:-/tmp}
```

Comment:
Having thought about this, I now realise that the host OS should have absolutely no responsibilities whatsoever. It should be entirely the transcoder's contract to ensure things are clean. So for example, there could be a pre-transcoding step that deletes pre-existing, un-deleted tmp files (that are over a certain age) and checks if there is enough disk space to even complete the transcoding.

With regards to the upcoming refactoring we're discussing, this will obviously fall under the transcoding tool. I'd like to see the transcoder work without docker, in fact I'd even like to see it refactored out from any S3 support -- that should be the user's responsibility. So this entire issue here will ultimately not be relevant.